src/components/Dropdown.tsx

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { sortingTypes } from "../App";
+import "../components/DropDown.css"
+import { CurrentSelection } from "./SortControl";
+
+interface IDropDown {
+    setCurrentSelection: (value: (((prevState: CurrentSelection) => CurrentSelection) | CurrentSelection)) => void
+}
+
+export default function Dropdown({ setCurrentSelection }: IDropDown) {
+    return (
+        <div className="dropDownContainer">
+            {sortingTypes.map(type => <button onClick={() => setCurrentSelection(type as CurrentSelection)}
```

Comment:
Looks like it makes sense to create 'sortingTypes'  from 'CurrentSelection' , so that you can avoid using type casting

Reply:
I just forgot to remove typecasting, there is no problem here with TS.